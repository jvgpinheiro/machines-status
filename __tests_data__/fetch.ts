type MockResponse = { json: () => []; text: () => "" };
type MockFetch = {
  resolve: (response: MockResponse) => void;
  reject: (reason: any) => void;
};

const fetchCallsToSolve = new Set<MockFetch>();

export function mockFetch(): Promise<MockResponse> {
  return new Promise((resolve, reject) => {
    fetchCallsToSolve.add({ resolve, reject });
  });
}

export function solveFetchs(): void {
  fetchCallsToSolve.forEach((call) =>
    call.resolve({ json: () => [], text: () => "" })
  );
  fetchCallsToSolve.clear();
}
