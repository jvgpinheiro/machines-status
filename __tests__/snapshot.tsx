import { render } from "@testing-library/react";
import Home from "../src/app/page";
import { mockFetch } from "../__tests_data__/fetch";

describe("Snapshot", () => {
  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => mockFetch());
  });

  it("renders homepage unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
