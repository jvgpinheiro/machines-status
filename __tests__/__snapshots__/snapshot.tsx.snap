// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders homepage unchanged 1`] = `
<div>
  <main
    class="main"
  >
    <nav
      class="navbar"
      data-testid="navbar-id"
    >
      <div
        class="navbarOption active"
        data-testid="navbar-home-id"
      >
        <span
          class="navbarText"
        >
          Home
        </span>
      </div>
      <div
        class="navbarOption "
        data-testid="navbar-assets-id"
      >
        <span
          class="navbarText"
        >
          Assets
        </span>
      </div>
      <div
        class="navbarOption "
        data-testid="navbar-users-id"
      >
        <span
          class="navbarText"
        >
          Users
        </span>
      </div>
      <div
        class="navbarOption "
        data-testid="navbar-units-id"
      >
        <span
          class="navbarText"
        >
          Units
        </span>
      </div>
      <div
        class="navbarOption "
        data-testid="navbar-workOrders-id"
      >
        <span
          class="navbarText"
        >
          Work Orders
        </span>
      </div>
    </nav>
    <div
      class="body"
      data-testid="body-id"
    >
      <div
        class="loadingContent"
        data-testid="loading-content-id"
      />
    </div>
  </main>
</div>
`;
